transformer_engine/pytorch/csrc/extensions/padding.cu

```diff
@@ -0,0 +1,79 @@
+/*************************************************************************
+ * Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See LICENSE for license information.
+ ************************************************************************/
+
+#include "extensions.h"
+
+void fused_multi_row_padding(at::Tensor input, at::Tensor output,
```

Comment:
Hi, why don't we simply pass a `std::vector<at::Tensor> input` so that we have all the size information and don't need to calculate the `d_input_ptr`, `d_output_ptr`?
This way was implemented in `te_grouped_gemm`.

Reply:
The call of `aten::split` has about 30us CPU overhead. We replace it by calculating pointer address of device in plain C++ to reduce CPU overhead. 
Ref from @yaox12 : https://github.com/NVIDIA/TransformerEngine/pull/1128/files#diff-342b0e9e5b472b443484b3a2c4a78647cd72431c272ed44c678ecbe636fb7a3aR191-R194

Reply:
Ah, I missed that PR. Thanks